Bluewire.Tools.Runner/FindCommits/ResolveCommitFromSemanticVersion.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Threading.Tasks;
+using Bluewire.Common.GitWrapper;
+using Bluewire.Common.GitWrapper.Model;
+using Bluewire.Conventions;
+using Bluewire.Tools.GitRepository;
+using System.Collections.Generic;
+using Bluewire.Tools.Runner.Shared;
+using System;
+
+namespace Bluewire.Tools.Runner.FindCommits
+{
+    public class ResolveCommitFromSemanticVersion : IBuildVersionResolutionJob
+    {
+        private readonly SemanticVersion semVer;
+
+        public ResolveCommitFromSemanticVersion(string semVer)
+        {
+            this.semVer = SemanticVersion.FromString(semVer);
+        }
+        
+        public async Task<Build[]> ResolveCommits(GitSession session, Common.GitWrapper.GitRepository repository)
+        {
+
+            // We expect only BuildNumberOutOfRangeException and BuildNumberNotFoundException exceptions.
+            // The caller of this method could notify the user that no builds were found but doesn't need the specifics.
+
+            if (semVer.IsComplete)
+            {
+                try
+                {
+                    var commit = await FindRemoteCommit(session, repository, semVer);

```

Comment:
Is this correct for the case where someone specifies `17.04.141-beta`, but the actual version of that commit lies in `17.05`?

IMO the 'completeness' of the semantic version is irrelevant, and the tag should act only as a filter on the results.